lib/index.js

```diff
@@ -130,4 +133,87 @@ Dynamoose.prototype.revertDDB = function () {
   this.dynamoDB = null;
 };
 
+Dynamoose.prototype.transaction = async function(items, options, next) {
+  debugTransaction('Run Transaction');
+  const deferred = Q.defer();
+
+  options = options || {};
+  if(typeof options === 'function') {
+    next = options;
+    options = {};
+  }
+
+  if(items === null || items === undefined || items.length === 0) {
+    deferred.reject(new errors.TransactionError('Items required to run transaction'));
+    return deferred.promise.nodeify(next);
+  }
+
+  items = await Promise.all(items);
+  let transactionReq = {
+    TransactItems: items.map(item => {
+      const returnItem = {...item};
+      delete returnItem.$__;
+      return returnItem;
+    })
+  };
+  let transactionMethodName = items.map(obj => Object.keys(obj)[0]).every(key => key === "Get") ? "transactGetItems" : "transactWriteItems";
```

Comment:
We should probably throw an error if the types differ between the requests - IIRC Dynamo will not support mixed read/write requests in the same transaction initially, thus if mixed `Get` and `Write` queries are passed in, it will break.

What about something like this:
```js
const operations = new Set(...items.map(obj => Object.keys(obj)[0]));

if (operations.length !== 1) {
  // Ensure same type of operations are provided.
  deferred.reject(new errors.TransactionError('Please use same query types - either "read" or "write" in the same transaction.'));
}

// Operations only have a single item at this point.
let transactionMethodName = operations.has('Get') ? "transactGetItems" : "transactWriteItems"
```

Reply:
@jutaz There are a lot of places I feel like throughout the code where we just rely on DynamoDB throwing the error. If DynamoDB ever decides to do something crazy and allow mixed queries, then no changes would be required to support that.